packages/frontend-2/lib/workspaces/composables/management.ts

```diff
@@ -417,6 +423,76 @@ export function useCreateWorkspace() {
   }
 }
 
+export const useWorkspaceSso = (params: { workspaceSlug: string }) => {
+  const { result, loading, error } = useQuery<WorkspaceSsoCheckQuery>(
+    workspaceSsoCheckQuery,
+    { slug: params.workspaceSlug }
+  )
+
+  const hasSsoEnabled = computed(() => !!result.value?.workspaceBySlug.sso?.provider)
+  const provider = computed(() => result.value?.workspaceBySlug.sso?.provider ?? null)
+
+  const isSsoAuthenticated = computed(() => {
+    return hasSsoEnabled.value && !needsSsoLogin.value
+  })
+
+  const needsSsoLogin = computed(() => {
+    if (!result.value?.activeUser) return false
+    return result.value.activeUser.expiredSsoSessions.some(
+      (workspace) => workspace.slug === params.workspaceSlug
+    )
+  })
+
+  return {
+    hasSsoEnabled,
+    isSsoAuthenticated,
+    needsSsoLogin,
+    provider,
+    loading,
+    error
+  }
+}
+
+export function useWorkspaceSsoPublic(workspaceSlug: string) {
```

Comment:
Some of these names could be improved. I've no idea what this composable does based on its name. And why is the loading only done on mounted, not in SSR?